src/data/finsupp/basic.lean

```diff
@@ -147,6 +147,12 @@ lemma ext_iff' {f g : α →₀ M} : f = g ↔ f.support = g.support ∧ ∀ x 
 ⟨assume h, ext $ assume a, by_contradiction $ λ H, (finset.ext_iff.1 h a).1 $
   mem_support_iff.2 H, by rintro rfl; refl⟩
 
+lemma support_nonempty_iff {f : α →₀ M} : f.support.nonempty ↔ f ≠ 0 :=
+by simp only [finsupp.support_eq_empty, finset.nonempty_iff_ne_empty, ne.def]
+
+lemma nonzero_iff_exists {f : α →₀ M} : f ≠ 0 ↔ ∃ a : α, f a ≠ 0 :=
+by simp [finsupp.support_eq_empty.symm, finset.eq_empty_iff_forall_not_mem]
```

Comment:
Do we have this statement somewhere?
```lean
lemma pi.nonzero_iff_exists {f : α → M} : f ≠ 0 ↔ ∃ a : α, f a ≠ 0 :=
sorry
```

Reply:
And also `add_monoid_hom.nonzero_iff_exists`, `monoid_hom.nonzero_iff_exists`, `ring_hom.nonzero_iff_exists`,  `linear_map.nonzero_iff_exists`, `alg_hom.nonzero_iff_exists`, ...

Reply:
This is #4985 all over again, of course.

Reply:
I do not know if these are there, but if you do not know about them, then I doubt it!

In any case, I do not know what the namespace `pi` stands for: is it a synonym for `set`?

Reply:
`pi` is the type of depedent functions - both of these can be thought of as pi types:
```
variables (α : Sort*) (β : α → Sort*) (γ : Sort*)
#check (Π a, β a)
#check (Π a : α, γ) -- aka α → γ
```
Mathlib is inconsistent when it comes to using the `pi` and `function` namespaces.

Reply:
Note that if you prove the `pi` version, `finsupp.nonzero_iff_exists` becomes simply `coe_fn_injective.eq_iff.trans the_pi_version`, possibly with some `.symm`s

Reply:
Eric, thank you for explaining the `pi` stuff!

While I understand that these `pi` lemmas should be there as well, I am not sure that I am willing to risk another growth of this PR like what happened in the past!  Would it be ok if I left it at that?  Maybe this issue can go in a `TODO` list?

Reply:
Definitely no need to grow this PR :)